### DISCUSSION TITLE
Sideloading Extensions

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#808] [Ideas]
### What feature would you like to see?

A few years back Firefox cut support for side loading your own custom extensions, I understand the security concerns around being able to load arbitrary extensions in the browser but it would be great if there was a properly supported system for super users to create their own custom extensions and then fully install them not just as a temporary extension.

### What platform are you seeing the problem on?

Linux

### Additional context

While there is support to load temporary extensions and you can use web-ext to run an ephemeral test browser during development I would like the ability not just to temporarily test my own extensions but to fully install them without having to get them approved by mozilla. 
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/524_

**Comment by Dekamir** [comment]
This doesn't have to be the default, but at least a config flag would be nice. This is the only reason I'm currently not using anything Firefox-based. Even Chrome allows you to sideload extensions without nagging you.
We shouldn't be limited to Mozilla's selection of extensions.

**Comment by mheidinger** [comment]
I second that: for work we have built an extension that is really useful for dev work but is something that we just can't publish to Mozilla for obvious reasons.
Being able to sideload them into Zen would be perfect, also fine if it is behind a dozen "are you sure" prompts or a config flag.

**Comment by shawny43** [comment]
Nah! Did you enable developer mode? I already being able to side-load an add-on last time i was tried to install an Discord Mods. Reason why it's a bit trash is because Firefox have bad behaviors to disable unverified addons after a bunch of weeks or months for security excuses, but again has i remember Vencord have been raging against Firefox because the approval system are just too problematic and complicated where in Chrome is not the case even it's a slow process validation, it's being able to validate their add-on. 

After all, to sideload in Firefox or any forks, you needs to enable developper mode before begin with it.

**Reply by shawny43** [reply]
Nevermind, you don't needs to enable developper mode. Just go in extension menu and click the gear. Install add-on and find your zip file.

**Comment by mheidinger** [comment]
After your message @shawny43 I had a look again and got it working! Although for my case it wasn't as straightforward.

Two pieces were missing:
1. My extensions needed to have an add-on ID in the manifest (v2 version). See [here](https://extensionworkshop.com/documentation/develop/extensions-and-the-add-on-id/) for details.
2. You need to disable the config setting `xpinstall.signatures.required`. Otherwise our local only extension can't be installed as it is not signed by Mozilla.

Hope that helps others aswell :)
